lib/showoff.rb

```diff
@@ -1375,13 +1375,16 @@ def valid_cookie
   end
 
   get %r{(?:image|file)/(.*)} do
-    path = params[:captures].first
-    full_path = File.join(settings.pres_dir, path)
-    if File.exist?(full_path)
-        send_file full_path
-    else
-        raise Sinatra::NotFound
-    end
+	path = params[:captures].first
+	full_path = File.join(settings.pres_dir, path)
+
+	if File.exist?(full_path)
+		send_file full_path
+	elsif File.exist?(path) # windows compatibility
```

Comment:
The path that comes from the request parameters is fully qualified on MS.


-------- Original message --------
From: Ben Ford <notifications@github.com>
Date: 5/10/16 7:49 PM (GMT-06:00)
To: puppetlabs/showoff <showoff@noreply.github.com>
Cc: "Tinn, Kevin" <kevin.tinn@asynchrony.com>, Author <author@noreply.github.com>
Subject: Re: [puppetlabs/showoff] Check if file exists at path or full path in image/file controller (#480)


In lib/showoff.rb<https://github.com/puppetlabs/showoff/pull/480#discussion_r62778187>:

> @@ -1375,13 +1375,16 @@ def valid_cookie
>    end
>
>    get %r{(?:image|file)/(.*)} do
> -    path = params[:captures].first
> -    full_path = File.join(settings.pres_dir, path)
> -    if File.exist?(full_path)
> -        send_file full_path
> -    else
> -        raise Sinatra::NotFound
> -    end
> +     path = params[:captures].first
> +     full_path = File.join(settings.pres_dir, path)
> +
> +     if File.exist?(full_path)
> +             send_file full_path
> +     elsif File.exist?(path) # windows compatibility


Is .pres_dir not working properly for you? I'll spin up a windows vm in a bit and validate that.

-
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub<https://github.com/puppetlabs/showoff/pull/480/files/be7fb4d1a0d00159154257f204a35835427aebb4#r62778187>